Problem2.py

```diff
@@ -0,0 +1,21 @@
+# 0-1 knapsack https://www.geeksforgeeks.org/0-1-knapsack-problem-dp-10/
+# Time complexity: O(n*m) => n = length of weights array, m = total target weight
+# Space complexity: O(n) since, second dimension is a constant (W+1)
```

Comment:
Second dimension isn't a constant. What if the items are super heavy and you have the capacity of 100,000?

SC would be O(n*m) too.